sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java

```diff
@@ -277,6 +349,95 @@ interface WithPublicNetworkAccess {
              * @return the next stage of the update
              */
             Update disablePublicNetworkAccess();
+
+            /**
+             * Specifies that by default access to container registry should be denied from all networks except from those
+             * networks specified via {@link DefinitionStages.WithPublicNetworkAccess#withAccessFromIpAddress(String)} ()}
+             * {@link WithPublicNetworkAccess#withAccessFromIpAddressRange(String)}.
+             *
+             * @return the next stage of the definition
+             */
+            Update withAccessFromSelectedNetworks();
+
+            /**
+             * Specifies that by default access to container registry should be allowed from all networks.
+             *
+             * @return the next stage of the definition
+             */
+            Update withAccessFromAllNetworks();
+
+            /**
+             * Specifies that access to the container registry from the specific ip range should be allowed.
+             * @param ipAddressCidr the ip address range expressed in cidr format
+             *
+             * @return the next stage of the definition
+             */
+            Update withAccessFromIpAddressRange(String ipAddressCidr);
+
+            /**
+             * Specifies that access to the container registry from the specific ip range should be denied.
+             *
+             * @param ipAddressCidr the ip address range expressed in cidr format
+             * @return the next stage of the definition
+             */
+            Update withoutAccessFromIpAddressRange(String ipAddressCidr);
+
+            /**
+             * Specifies that access to the container registry from the specific ip address should be allowed.
+             *
+             * @param ipAddress the ip address
+             * @return the next stage of the definition
+             */
+            Update withAccessFromIpAddress(String ipAddress);
+
+            /**
+             * Specifies that access to the container registry from the specific ip address should be denied.
```

Comment:
Maybe it better to say "remove the allowed ip address"?

Since else user might think to "with" a range, then "without" a few address within that range. I assume this is not possible?

Reply:
Yeah, remove is better, since it's a remove action, not a blacklist.